src/components/modals/forms/RegistrationForm.tsx

```diff
@@ -43,21 +42,19 @@ export default function RegistrationForm() {
       setLoading(true)
       const data = await apiRegisterUser(formData)
 
-      authenticate(data.token)
-      reset()
-      redirectToPreviousRoute()
-    } catch (e) {
-      if (e instanceof ServerError) {
-        setError('root.serverError', {
-          type: '500',
-          message: e.message,
-        })
-      } else {
-        setError('email', {
-          type: 'manual',
-          message: 'This email already exists',
-        })
+      if (data) {
```

Comment:
this check is done for safety purposes, since the code setRegistrationButtonDisabled(true)
        router.push('/auth/registration/confirm_password') is synchronous and there can be a situation when an error will be triggered asynchronously and our previous code will work and let us through the route! That is, something will go wrong during registration and we will be on the password confirmation page! in java script there is such a concept as event loop which divides into stacks synchronous and asynchronous code and first of all synchronous code triggers and after it in the order of turn triggers promises and the rest of asynchronous code! it is connected with the fact that the js language itself is single-threaded and for this they invented such a supposed emulator !I think this check is not superfluous